wdl-lint/src/rules/description_missing.rs

```diff
@@ -0,0 +1,91 @@
+//! A lint rule to ensure a description is included in `meta` sections.
+
+use wdl_ast::v1::MetadataSection;
+use wdl_ast::AstNode;
+use wdl_ast::AstToken;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Document;
+use wdl_ast::Span;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the description missing rule.
+const ID: &str = "DescriptionMissing";
+
+/// Creates a description missing diagnostic.
+fn description_missing(span: Span) -> Diagnostic {
+    Diagnostic::note("this section is missing a description key")
```

Comment:
Potential edit:
```suggestion
    Diagnostic::note("<task or workflow `name`> missing a description meta key")
```
I'm thinking of how it will appear in short-form. `this` loses context when all you have is the line+col number. The name of the task or workflow would be nice to have in a context like `Arena`.

Reply:
It has a `highlight` which tags the relevant `meta` keyword.

Reply:
Nevermind @a-frantz , I see what you mean. Now I'm reporting like this: 
```
note[DescriptionMissing]: task `foo` is missing a description key
  ┌─ tests/lints/description-missing/source.wdl:6:5
  │
6 │     meta {}
  │     ^^^^
  │
  = fix: add a `description` key to this meta section
```

Reply:
Nice, LGTM 👍